### ISSUE TITLE
Avoid recomputing dimensions/offer option to force recompute

### ISSUE THREAD

**Issue by chambbj** [PDAL/PDAL#2057]
There are at least a handful of places where we compute a value that may have been computed by an upstream stage and already exist as a Dimension. This one comes to mind

https://github.com/PDAL/PDAL/blob/6e2d2a7363b2969b6e895d21b7e8381978cad7cf/filters/ApproximateCoplanarFilter.cpp#L93-L97

which at least conceivably could have been computed (and set) here

https://github.com/PDAL/PDAL/blob/6e2d2a7363b2969b6e895d21b7e8381978cad7cf/filters/EigenvaluesFilter.cpp#L92-L100

We should use the existing Dimension where possible, unless the user specifies that they want to force a recompute in the stage options.

**Comment by abellgithub** [comment]
I don't think any infrastructure is needed for this.  It's just about agreeing on dimensions.  I guess we just need to be sure to describe the dimensions used with sufficient detail.

**Comment by chambbj** [comment]
No infrastructure changes per se, but handled on a case-by-case basis. So in the example, the check and possible recompute would all be handled within `filters.approximatecoplanar`.

**Comment by abellgithub** [comment]
Just trying to think of the canonical handling.  Should a downstream stage that sees a dimension it would otherwise compute is already in the layout assume it's going to be computed?  Or should some metadata be passed?  Something else?

**Comment by chambbj** [comment]
Right. I guess my assumption had been that certain dimensions exist solely because they have been computed upstream (like the eigenvalues). But that doesn't mean somebody couldn't have imported data with an empty or invalid dimension that needs to be computed. Still, if this is the case, then the user has to own that. Whether they insert `filters.eigenvalues` or set `--force` to recompute is up to them.
